src/ui/components/DropdownMenuItem/index.js

```diff
@@ -0,0 +1,57 @@
+/* @flow */
+import React from 'react';
+import classNames from 'classnames';
+import { oneLine } from 'common-tags';
+
+import Link from 'amo/components/Link';
+
+import './styles.scss';
+
+
+type Props = {|
+  children: string | Link,
+  detached?: boolean,
+  onClick?: Function,
+|};
+
+const DropdownMenuItem = ({ children, onClick, detached = false }: Props) => {
+  const childIsComponent = typeof children === 'object';
+
+  if (
+    childIsComponent &&
+    children.type && children.type.displayName !== 'Connect(LinkBase)'
+  ) {
+    throw new Error(oneLine`Only the "Link" component is supported as a child
```

Comment:
My initial idea was to restrict the menu to `Link` or `text` entries. There are Flow types but it will not be caught because Flow is not used everywhere yet. That is why I have added such a check (semi-related: https://github.com/mozilla/addons-frontend/pull/2972#discussion_r134330441).

Maybe it is not needed though.

Reply:
I think there's nothing technically preventing another element from being a child so this check seems unneeded. I get that it might be awkward to add a non-linked element in there but I say we remove it for now.